src/lib/ares_private.h

```diff
@@ -248,6 +248,8 @@ struct query {
   int using_tcp;
   int error_status;
   int timeouts; /* number of timeouts we saw for this request */
+  int no_retries; /* do not perform any additional retries, this is set when
```

Comment:
Maybe make it `no_retries:1;` to signal that it actually is a boolean? Alternatively use a bool typedef.

Reply:
This is another one where "convention" within c-ares gets in the way.  There is no `bool` type defined or used anywhere within c-ares, and I can't find any other instances of using the bitfield width modifiers.  You can see the "using_tcp" member above, it too is a boolean, however it too uses no modifiers.  I'd rather keep consistency at this point and maybe do a bulk "cleanup" in the future for things like this.